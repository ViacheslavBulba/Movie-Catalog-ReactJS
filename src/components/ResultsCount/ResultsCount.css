.result-count-container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin-bottom: 20px;
    margin-left: 60px;
    margin-right: 60px;
}
