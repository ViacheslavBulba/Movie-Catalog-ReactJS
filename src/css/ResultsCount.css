.result-count-container {
    margin-top: 20px;
    margin-bottom: 20px;
    margin-left: 60px;
}
