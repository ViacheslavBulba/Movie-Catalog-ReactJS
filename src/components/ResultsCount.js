
import React from 'react';
import '../css/ResultsCount.css';
import PropTypes from 'prop-types';

export default function ResultsCount(props) {

    return (
        <div className="result-count-container">
            <b>{props.count}</b> movies found
        </div>
    );
}

ResultsCount.propTypes = {
    count: PropTypes.number.isRequired,
};
