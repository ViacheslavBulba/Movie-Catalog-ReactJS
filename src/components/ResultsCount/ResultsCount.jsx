import React from 'react';
import './ResultsCount.css';
import { useSelector } from 'react-redux';

export default function ResultsCount() {
    const count = useSelector((state) => state.movies.length);

    return (
        <>
            {count !== 0 && (
                <div className='result-count-container'>
                    <b>{count}</b> movies found
                </div>
            )}
        </>
    );
}
