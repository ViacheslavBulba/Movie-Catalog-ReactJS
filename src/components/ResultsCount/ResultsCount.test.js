import React from 'react';
import renderer from 'react-test-renderer';
import ResultsCount from './ResultsCount';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

const initialState = {
    pending: false,
    movies: [
        {
            'id': 1,
            'title': 'mock movie',
            'tagline': 'mock tagline',
            'vote_average': 9.9,
            'vote_count': 0,
            'release_date': '2021-04-18',
            'poster_path': 'https://image.tmdb.org/t/p/w500/3kcEGnYBHDeqmdYf8ZRbKdfmlUy.jpg',
            'overview': 'mock overview',
            'budget': 1000,
            'revenue': 2000,
            'genres': ['Drama', 'Romance'],
            'runtime': 90
        }
    ],
    error: null,
    sortBy: 'release_date',
    sortOrder: 'desc',
    filerByGenres: [],
    search: '',
    searchBy: 'title',
    movieToOverview: null,
};

const store = mockStore(initialState);

test('snapshot test for results count', () => {
    const component = renderer.create(
        <Provider store={store}>
            <ResultsCount />
        </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
