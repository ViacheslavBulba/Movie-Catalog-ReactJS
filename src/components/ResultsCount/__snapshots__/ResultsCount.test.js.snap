// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for results count 1`] = `
<div
  className="result-count-container"
>
  <b>
    1
  </b>
   movies found
</div>
`;
